pkg/daemon/ceph/client/pool.go

```diff
@@ -303,29 +306,38 @@ func CreateECPoolForApp(context *clusterd.Context, clusterInfo *ClusterInfo, poo
 	return nil
 }
 
-func CreateReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterInfo, poolName string, pool cephv1.PoolSpec, pgCount, appName string, cluster cephv1.ClusterSpec) error {
-	// Create a CRUSH rule for stretched clusters
-	if pool.Replicated.ReplicasPerFailureDomain != 0 {
-		err := createStretchedReplicationCrushRule(context, clusterInfo, poolName, pool, cluster)
-		if err != nil {
-			return errors.Wrap(err, "failed to create stretched replicated crush rule")
-		}
+func CreateReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, poolName string, pool cephv1.PoolSpec, pgCount, appName string) error {
+	// The crush rule name is the same as the pool unless we have a stretch cluster.
+	crushRuleName := poolName
+	if clusterSpec.IsStretchCluster() {
```

Comment:
I'm not sure I understand this condition. If not stretch cluster we call `createStretchedReplicationCrushRule`? And if stretch we do nothing? 🤔

Reply:
- If it's a stretch cluster we will always use the same crush rule for all pools. 
- If it's not a stretch cluster, we will go ahead and create the crush rule. If they set ReplicasPerFailureDomain, it's not just for stretch clusters, so i'll rename that helper.

Reply:
Oh I see so the `defaultStretchCrushRuleName` is created somewhere else? More comments in the code about the behavior would be nice.
It's just weird to not see the CRUSH rule creation in this code, it gives the impression we do nothing.

Reply:
Shouldn't we use `crushRuleName` instead `poolName` in `createTwoStepCrushRule()`?

Reply:
Yes, I'll add more comments around the behavior how the crush rule is created elsewhere, and the same crush rule is used for all pools in a stretch cluster. And yes, I'll update to use `crushRuleName` instead of `poolName`.

Reply:
Ok great!